### PR TITLE
Do not explode on PHP files

### LINKED ISSUE #290: [ Bug ] HHAST tries to lint `<?php` files
If HHAST sees a `.php` file it is instructed to lint, but this is a `<?php` file, this will almost always throw a fatal error, since `[]` array creation is illegal in Hack.

HHAST should skip `.php` files that do not contain a `<?hh` token.

### PR DESCRIPTION
**Pull request by lexidor**
closes #290

### CODE REVIEW

**Comment by lexidor** [comment]
This change isn't very pretty, but it prevents hhast from going kaboom when it hits a `.php` file on accident that actually contains php.
I can lint in vscode-mode (tested by making a method in a final class final and seeing the yellow squigle) and using `bin/hhast-lint`, see travis output.

**Comment by lexidor** [comment]
The :x: on FAILS-LINT is intentional.
It is the test case to show that `isHackFile()` is actually guarding the parser correctly.

I added `.vscode/**` to export-ignore, because I don't know if hhvm-autoload will try and parse the gunk php file when `composer require hhvm/hhast --dev`.

**Comment by lexidor** [comment]
If I put it in a folder that hhvm-autoload can see, it will not be able to generate an autoloadmap.
It fails with a parse error.
I can place this anywhere, except for `src` and `tests`.
The directory needs to be added to `hhast-lint.json` to test effectively.

**Comment by fredemmott** [comment]
can you add a `test-data/` top-level directory or similar?

File: hhast-lint.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "roots": [ "src/", "codegen/", "tests/" ],
+  "roots": [ "src/", "codegen/", "tests/", "test-data/" ],
```
Comment (fredemmott) [inline hhast-lint.json:2]: Please remove this and add a test that invokes this and makes sure it's not linted

Comment (lexidor) [reply hhast-lint.json]: Didn't see this earlier, resolved.

**Pull request merged**